### ISSUE TITLE
NVDA, internet explorer and Amazon.

### ISSUE THREAD

**Issue by chris0309** [nvaccess/nvda#6022]
Hi, 
There is a slight issue with NVDA on the Amazon website using internet explorer. 
It is very difficult to click on links on the Amazon site. The only way to click on links is to tab to the link you are looking for. It is the same way with certain buttons. 
I have noticed this with the past few versions of NVDA. I think the first time I noticed it was in NVDA 2015.3. 
Here are the steps I take. 
1. I go to the Amazon website. 
2. I go to the edit field and type in a search query.
3.  When the results come up, I move by heading by pressing the letter H. When I get to the link I want, I hit enter, nothing happens. 
4. I go back to the top of the page, and tab until I hear the link or button I want. I click enter, and the product page comes up. 
I just wanted to make you aware of this. 
Thanks 
Chris

**Comment by Adriani90** [comment]
@chris0309 are you still able to reproduce this issue with NVDA 2018.4.1 or with last NVDA Alpha? Thanks!

**Comment by Adriani90** [comment]
On Amazon and IE NVDA seems to fail when trying to open a link or to activate a button almos anywhere on the webpage. In browse mode, when using arrow keys to navigate and enter key to actiave elements, almost everything fails. This is actually really critical because IE is being used still in many cases. cc: @JulienCochuyt, @feerrenrut

**Comment by JulienCochuyt** [comment]
I do not reproduce the faulty scenario described in https://github.com/nvaccess/nvda/issues/6022#issue-157569968 browsing with IE 11.615.17763.0 on Windows 10 and installed NVDA 2019.2beta3 nor portable NVDA 2018.4.

After performing the search and hitting "H" a couple of times to get to a random result, hitting enter does activate the link.
Alternatively, pressing the down arrow a few times to go from one result to the next one and hitting enter once again activate the link.

Nevertheless, performance is really poor and I have to carefully wait for a proper reaction between each and every key press to ensure the whole thing does not break appart - despite testing on a race-horse workstation.

**Comment by Adriani90** [comment]
In some examples works for me as well to activate the links but not everytime. Often I get following errors in the log:

```
IO - inputCore.InputManager.executeGesture (17:36:30.828):
Input: kb(laptop):enter
DEBUGWARNING - virtualBuffers.VirtualBuffer._isNVDAObjectInApplication_noWalk (17:36:32.013):
getIdentifierFromNVDAObject failed. Object probably died while walking ancestors.
Traceback (most recent call last):
  File "virtualBuffers\__init__.pyc", line 703, in _isNVDAObjectInApplication_noWalk
  File "virtualBuffers\MSHTML.pyc", line 231, in getIdentifierFromNVDAObject
LookupError
IO - inputCore.InputManager.executeGesture (17:36:35.226):
Input: kb(laptop):upArrow
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:36:36.739):
accRole failed: (-2147467259, 'Unbekannter Fehler', (None, None, None, 0, None))
DEBUG - treeInterceptorHandler.killTreeInterceptor (17:36:36.739):
Killed treeInterceptor: <virtualBuffers.MSHTML.MSHTML object at 0x04DB2710>
DEBUG - treeInterceptorHandler.update (17:36:36.786):
Adding new treeInterceptor to runningTable: <virtualBuffers.MSHTML.MSHTML object at 0x04F236D0>
DEBUG - virtualBuffers.VirtualBuffer._loadBuffer (17:36:38.086):
Buffer load took 1.300 sec, 18273 chars
ERROR - scriptHandler.executeScript (17:36:39.859):
error executing script: <bound method MSHTML.script_moveByLine_back of <virtualBuffers.MSHTML.MSHTML object at 0x04DB2710>> with gesture u'up arrow'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 192, in executeScript
  File "cursorManager.pyc", line 253, in script_moveByLine_back
  File "cursorManager.pyc", line 158, in _caretMovementScriptHelper
  File "documentBase.pyc", line 24, in makeTextInfo
  File "virtualBuffers\__init__.pyc", line 207, in __init__
  File "textInfos\offsets.pyc", line 407, in __init__
  File "virtualBuffers\__init__.pyc", line 212, in _getSelectionOffsets
WindowsError: [Error 1775] Ein NULL-Kontexthandle wurde während eines Remoteprozeduraufrufs vom Client an den Host übergeben
IO - inputCore.logTimeSinceInput (17:36:42.132):
6.907 sec since input
IO - speech.speak (17:36:42.132):
Speaking [u'LARNMERN Stahlkappe Sicherheitsschuhe, Herren luftdurchl\xe4ssige Leichte Anti-Smashing Schuhe Industrie und Handwerk\uff0cLM-123K (46 EU, Reflektierendes Blau): Amazon.de: Amazon.de  document']
```
cc: @leonardder

**Comment by Adriani90** [comment]
Especially after the windows error, the performance in IE is extremely bad. About 6 seconds between key press and speech. I am on Windows 10 1903 and IE 11.239 64bit.
cc: @lukaszgo1, @k-kolev1985 could you please test on Windows 7 and Windows 8.1 if this occurs for you and how the performance is?

**Comment by JulienCochuyt** [comment]
@Adriani90, I realize you are on a German locale: Which Amazon site are you testing?
I'm asking because I'm getting different results depending on the country.
amazon.com and amazon.fr give poor performance, but are still quite usable on my side.
amazon.de, on the other hand, is almost unusable from here: I even had a pretty hard time to bring the focus in the search field.

**Comment by Adriani90** [comment]
I was testing on amazon.de. Actually this is really strange that we get different results depending on locale. In Firefox or Chrome this is not the case.

**Comment by Adriani90** [comment]
I am closing this issue since Support for Internet explorer has ended in 2022 and the IE mode in Edge for backwards compatibility will most probably not get accessibility updates from any side.
In Edge, Firefox and Chrome everything works as expected.
